cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-westend/src/tests/snowbridge_v2_inbound_to_rococo.rs

```diff
@@ -0,0 +1,616 @@
+
+	AssetHubRococo::execute_with(|| {
+		type RuntimeEvent = <AssetHubRococo as Chain>::RuntimeEvent;
+
+		assert_expected_events!(
+			AssetHubRococo,
+			vec![
+				// Message processed successfully
+				RuntimeEvent::MessageQueue(
+					pallet_message_queue::Event::Processed { success: true, .. }
+				) => {},
+				// Ether was deposited to beneficiary
+				RuntimeEvent::ForeignAssets(pallet_assets::Event::Issued { asset_id, owner, .. }) => {
+					asset_id: *asset_id == eth_location(),
+					owner: *owner == beneficiary_acc_bytes.into(),
+				},
+			]
+		);
+
+		let events = AssetHubRococo::events();
+		// Check that no assets were trapped
+		assert!(
+			!events.iter().any(|event| matches!(
+				event,
+				RuntimeEvent::PolkadotXcm(pallet_xcm::Event::AssetsTrapped { .. })
+			)),
+			"Assets were trapped on Rococo AssetHub, should not happen."
+		);
+	});
+}
+
+#[test]
+fn send_roc_from_ethereum_to_rococo() {
+	let initial_fund: u128 = 200_000_000_000_000;
+	let initial_liquidity: u128 = initial_fund / 2;
+	let amount: u128 = initial_fund;
+	let roc_fee_amount: u128 = initial_liquidity / 2;
+	let wnd_amount_to_swap: u128 = initial_liquidity / 10;
+
+	let relayer_account = BridgeHubWestendSender::get();
+	let relayer_reward = 1_500_000_000_000u128;
+
+	let claimer = AccountId32 { network: None, id: H256::random().into() };
+	let claimer_bytes = claimer.encode();
+	let beneficiary =
+		Location::new(0, AccountId32 { network: None, id: AssetHubRococoReceiver::get().into() });
+
+	BridgeHubWestend::fund_para_sovereign(AssetHubWestend::para_id(), INITIAL_FUND);
+
+	let ethereum_sovereign: AccountId = snowbridge_sovereign();
+	let sender = AssetHubRococoSender::get();
+	let roc_at_asset_hub_rococo = roc_at_ah_rococo();
+	let bridged_roc_at_asset_hub_westend = bridged_roc_at_ah_westend();
+	create_foreign_on_ah_westend(bridged_roc_at_asset_hub_westend.clone(), true);
+	set_up_pool_with_wnd_on_ah_westend(
+		bridged_roc_at_asset_hub_westend.clone(),
+		true,
+		initial_fund,
+		initial_liquidity,
+	);
+
+	BridgeHubRococo::fund_para_sovereign(AssetHubRococo::para_id(), initial_fund);
+	AssetHubRococo::fund_accounts(vec![(AssetHubRococoSender::get(), initial_fund)]);
+	fund_on_bh();
+	register_roc_on_bh();
+
+	set_up_eth_and_dot_pool();
+	set_up_eth_and_dot_pool_on_rococo();
+
+	// set XCM versions
+	BridgeHubWestend::force_xcm_version(asset_hub_westend_location(), XCM_VERSION);
+	BridgeHubWestend::force_xcm_version(asset_hub_rococo_location(), XCM_VERSION);
+	AssetHubWestend::force_xcm_version(asset_hub_rococo_location(), XCM_VERSION);
+
+	let local_fee_asset: Asset = (roc_at_asset_hub_rococo.clone(), roc_fee_amount).into();
+	let assets: Assets = (roc_at_asset_hub_rococo.clone(), amount).into();
+	let asset: Asset =
+		(roc_at_asset_hub_rococo.clone(), amount - roc_fee_amount * 2).into();
+	let roc_at_asset_hub_rococo_reanchored: Asset =
+		(bridged_roc_at_asset_hub_westend.clone(), (amount - roc_fee_amount * 2) / 2).into();
```

Comment:
These variables seems unused? I'd suggest some furthur cleanup.

Also, please check the code format - it doesn't seem correct.

```
cargo +nightly fmt --all -- --check
```

Reply:
https://github.com/vgeddes/polkadot-sdk/pull/58/commits/8303bfebb812ef724d3e844c07fadb8b84e22d39 :)